lib/packwerk/offense_collection.rb

```diff
@@ -46,9 +50,13 @@ def add_offense(offense)
         @errors << offense
         return
       end
-      package_todo = package_todo_for(offense.reference.package)
-      unless package_todo.add_entries(offense.reference, offense.violation_type)
+
+      already_listed = package_todo_for(offense.reference.package).add_entries(offense.reference,
+        offense.violation_type)
+      if !already_listed
```

Comment:
Let's make this a private method too for readability.

Reply:
sounds good